activerecord/test/cases/validations/numericality_validation_test.rb

```diff
@@ -177,4 +177,60 @@ def test_allow_nil_works_for_casted_value
 
     assert_predicate subject, :valid?
   end
+
+  MAX_EIGHT_BYTES_SIGNED_NUMBER = 2**63 - 1
+  private_constant :MAX_EIGHT_BYTES_SIGNED_NUMBER
+
+  def test_column_in_limit
+    update_limit_for_column(model_class, "bank_balance", new_limit: 8)
+    model_class.validates_numericality_of(:bank_balance, in: :limit)
+
+    subject = model_class.new(bank_balance: MAX_EIGHT_BYTES_SIGNED_NUMBER + 1)
+
+    assert_not_predicate subject, :valid?
+
+    subject = model_class.new(bank_balance: MAX_EIGHT_BYTES_SIGNED_NUMBER)
+
+    assert_predicate subject, :valid?
+  end
+
+  def test_column_in_limit_without_limit_is_ignored
+    model_class.validates_numericality_of(:bank_balance, in: :limit)
+
+    subject = model_class.new(bank_balance: MAX_EIGHT_BYTES_SIGNED_NUMBER + 1)
+
+    assert_predicate subject, :valid?
+  end
+
+  def test_virtual_column_in_limit_raises
+    model_class.attribute(:virtual_bank_balance, :decimal)
+    model_class.validates_numericality_of(:virtual_bank_balance, in: :limit)
+
+    subject = model_class.new(virtual_bank_balance: MAX_EIGHT_BYTES_SIGNED_NUMBER + 1)
+
+    assert_raises ArgumentError, "cannot validate :limit for a virtual attribute" do
+      subject.valid?
+    end
+  end
+
+  private
+    def update_limit_for_column(model_class, attr_name, new_limit:)
```

Comment:
Since this is an Active Record test, I think it would be better if you actually dropped and recreated the columns (or just made a new column for each test and dropped it in an `ensure`), that way you have a more realistic test case.

Reply:
https://github.com/rails/rails/blob/main/activerecord/test/cases/date_time_precision_test.rb has lots of examples. Many tests create a `:foos` table and it's dropped in `teardown`.